src/com/ms/silverking/cloud/dht/management/aws/ExampleTestCode.java

```diff
@@ -19,109 +19,95 @@
-                    e);
-        }
-        ec2 = AmazonEC2ClientBuilder.standard()
-            .withRegion("us-west-2")
-            .build();
-    }
-    
-    public void run() {
-        try {
-            DescribeAvailabilityZonesResult availabilityZonesResult = ec2.describeAvailabilityZones();
-            System.out.println("You have access to " + availabilityZonesResult.getAvailabilityZones().size() +
-                    " Availability Zones.");
-
-            DescribeInstancesResult describeInstancesRequest = ec2.describeInstances();
-            List<Reservation> reservations = describeInstancesRequest.getReservations();
-            Set<Instance> instances = new HashSet<Instance>();
-
-            System.out.println("Reserves size: " + reservations.size());
-            for (Reservation reservation : reservations) {
-                instances.addAll(reservation.getInstances());
-            }
-
-            System.out.println("You have " + instances.size() + " Amazon EC2 instance(s) running.");
-        } catch (AmazonServiceException ase) {
-                System.out.println("Caught Exception: " + ase.getMessage());
-                System.out.println("Reponse Status Code: " + ase.getStatusCode());
-                System.out.println("Error Code: " + ase.getErrorCode());
-                System.out.println("Request ID: " + ase.getRequestId());
-        }
+  private AmazonEC2 ec2;
+  private static final String amiId = "ami-b77b06cf";
+  private static final String keyPairName = "bph";
+  private static final String securityGroupName = "bph";
+
+  public ExampleTestCode() {
+    /*
+     * The ProfileCredentialsProvider will return your [default]
+     * credential profile by reading from the credentials file located at
+     * (C:\\Users\\ben-pc\\.aws\\credentials).
+     */
+    ProfileCredentialsProvider credentialsProvider = new ProfileCredentialsProvider();
+    try {
+      credentialsProvider.getCredentials();
+    } catch (Exception e) {
+      throw new AmazonClientException(
+          "Cannot load the credentials from the credential profiles file. " + "Please make sure that your credentials" +
+              " file is at the correct " + "location (C:\\Users\\ben-pc\\.aws\\credentials), and is in valid format.",
+          e);
     }
-    
-    public void runInstances() {
-        RunInstancesRequest runInstancesRequest = new RunInstancesRequest();
-        runInstancesRequest.withImageId(amiId)
-                           .withInstanceType("t2.micro")
-                           .withMinCount(1)
-                           .withMaxCount(1)
-                           .withKeyName(keyPairName)
-                           .withSecurityGroups(securityGroupName);
-                
-        RunInstancesResult result = ec2.runInstances(runInstancesRequest);
+    ec2 = AmazonEC2ClientBuilder.standard().withRegion("us-west-2").build();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

AWS Region is set using a String. To explicitly set a publicly available region, we recommend that you use the [Regions](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/index.html?com/amazonaws/regions/Regions.html) enum.

[Learn more](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/java-dg-region-selection.html#region-selection-choose-region)